rkt/metadata_service.go

```diff
@@ -541,12 +546,20 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
+			stderr("Error serving registration HTTP: %v", err)
+			cleanup()
+			os.Exit(1)
+		}
+	}()
+
+	if err = srv.Serve(tcpl); err != nil {
```

Comment:
I really don't like this, it's still kinda racy with both of them potentially calling cleanup() and stomping on each other. Can't you just treat the two serves as equal, start them both in a goroutine and use a channel to act on the first error?